includes/class-simple-local-avatars.php

```diff
@@ -694,4 +712,100 @@ private function update_avatar_ratings() {
 			'X'  => __( 'X &#8212; Even more mature than above' ),
 		);
 	}
+
+	/**
+	 * Load script required for handling any actions.
+	 */
+	public function admin_scripts() {
+		wp_enqueue_script(
+			'sla_admin',
+			SLA_PLUGIN_URL . 'assets/js/admin.js',
+			[ 'jquery' ],
+			SLA_VERSION,
+			true
+		);
+
+		wp_localize_script(
+			'sla_admin',
+			'slaAdmin',
+			[
+				'nonce' => wp_create_nonce( 'sla_clear_cache_nonce' ),
+				'error' => esc_html__( 'Something went wrong while clearing cache, please try again.', 'simple-local-avatars' ),
+			]
+		);
+	}
+
+	/**
+	 * Clear user cache.
+	 */
+	public function sla_clear_user_cache() {
+		check_ajax_referer( 'sla_clear_cache_nonce', 'nonce' );
+		$step = isset( $_REQUEST['step'] ) ? intval( $_REQUEST['step'] ) : 1;
+
+		// Setup defaults.
+		$users_per_page = 50;
+		$offset         = ( $step - 1 ) * $users_per_page;
+
+		$rows        = array();
+		$users_query = new \WP_User_Query(
+			array(
+				'fields' => array( 'ID' ),
+				'number' => $users_per_page,
+				'offset' => $offset,
+			)
+		);
+
+		// Get the users.
+		$users = $users_query->get_results();
+
+		if ( ! empty( $users ) ) {
+			foreach ( $users as $user ) {
+				$user_id       = $user->ID;
+				$rows[]        = $user_id;
+				$local_avatars = get_user_meta( $user_id, 'simple_local_avatar', true );
+				$this->clear_user_avatar_cache( $local_avatars, $user_id, $local_avatars['media_id'] ?? '' );
+			}
+		}
+
+		if ( ! empty( $rows ) ) {
```

Comment:
Instead of creating `$row` to hold only user IDs, I think we can use the `$users` variable above to check for the end of the progress.

Reply:
@dinhtungdu I've updated this, please check again when you get a chance, thanks!